python_modules/dagster/dagster/_core/remote_representation/code_location.py

```diff
@@ -428,55 +445,53 @@ def get_external_execution_plan(
     def get_external_partition_config(
         self,
         repository_handle: RepositoryHandle,
-        partition_set_name: str,
+        job_name: str,
         partition_name: str,
         instance: DagsterInstance,
     ) -> Union["ExternalPartitionConfigData", "ExternalPartitionExecutionErrorData"]:
         check.inst_param(repository_handle, "repository_handle", RepositoryHandle)
-        check.str_param(partition_set_name, "partition_set_name")
+        check.str_param(job_name, "job_name")
         check.str_param(partition_name, "partition_name")
 
         return get_partition_config(
             self._get_repo_def(repository_handle.repository_name),
-            partition_set_name=partition_set_name,
+            job_name=job_name,
             partition_key=partition_name,
             instance_ref=instance.get_ref(),
         )
 
     def get_external_partition_tags(
         self,
         repository_handle: RepositoryHandle,
-        partition_set_name: str,
+        job_name: str,
         partition_name: str,
         instance: DagsterInstance,
+        selected_asset_keys: Optional[AbstractSet[AssetKey]],
     ) -> Union["ExternalPartitionTagsData", "ExternalPartitionExecutionErrorData"]:
         check.inst_param(repository_handle, "repository_handle", RepositoryHandle)
-        check.str_param(partition_set_name, "partition_set_name")
+        check.str_param(job_name, "job_name")
         check.str_param(partition_name, "partition_name")
         check.inst_param(instance, "instance", DagsterInstance)
 
         return get_partition_tags(
             self._get_repo_def(repository_handle.repository_name),
-            partition_set_name=partition_set_name,
+            job_name=job_name,
             partition_name=partition_name,
             instance_ref=instance.get_ref(),
+            selected_asset_keys=selected_asset_keys,
         )
 
     def get_external_partition_names(
-        self, external_partition_set: ExternalPartitionSet, instance: DagsterInstance
+        self,
+        repository_handle: RepositoryHandle,
+        job_name: str,
+        instance: DagsterInstance,
+        selected_asset_keys: Optional[AbstractSet[AssetKey]],
     ) -> Union["ExternalPartitionNamesData", "ExternalPartitionExecutionErrorData"]:
-        check.inst_param(external_partition_set, "external_partition_set", ExternalPartitionSet)
-
-        # Prefer to return the names without calling out to user code if the
-        # partition set allows it
-        if external_partition_set.has_partition_name_data():
-            return ExternalPartitionNamesData(
-                partition_names=external_partition_set.get_partition_names(instance)
-            )
```

Comment:
This seems like an optimization that could have big implications if it was being used a lot and was removed (specifically, a lot more calls to user code). Although the fact that it is being called on the InProcessCodeLocation is confusing since you're sort of already in user code at that point (and InProcessCodeLocation isn't really used outside of tests and maybe the CLI), so maybe a moot point

Reply:
linking this here for future readers: https://github.com/dagster-io/dagster/pull/23491/files#r1715290754